### PR TITLE
MD-75 Estudar adição do redux para gerenciamento de estado

### PR DESCRIPTION
**Pull request by EricksonSiqueira**

### CODE REVIEW

File: src/components/ProductsWithSearchInput/index.tsx [hunk 1]
```diff
@@ -1,50 +1,18 @@
...
-  };
-
-  const editProductFromList = (newProduct: Product) => {
-    const newProductsList = products.map((product) =>
-      product.id === newProduct.id ? newProduct : product
-    );
-
-    setProducts(newProductsList);
-  };
+  const reduxProducts = useAppSelector(
```
Comment (claudiocassimiro) [inline src/components/ProductsWithSearchInput/index.tsx:13]:
```suggestion
  const products = useAppSelector(
```
I think just products is a good name

File: src/components/ProductsWithSearchInput/index.tsx [hunk 2]
```diff
@@ -55,16 +23,14 @@ export default function ProductsWithSearchInput({
         value={searchInputValue}
       />
       <div className={styles.productsWithSearchInputContainerProducts}>
-        {products.length > 0 ? (
-          products
+        {reduxProducts.length > 0 ? (
+          reduxProducts
```
Comment (claudiocassimiro) [inline src/components/ProductsWithSearchInput/index.tsx:27]:
```suggestion
        {products.length > 0 ? (
          products
```

File: src/components/YourStore/index.tsx [hunk 3]
```diff
@@ -13,58 +12,14 @@ export interface YourStoreProps {
   categories: GetCategoryReturn[];
 }
 
-export default function YourStore({ categories }: YourStoreProps) {
-  const [categoriesWithProducts, setCategoriesWithProducts] =
-    useState(categories);
-  const onlyCategoriesWithProducts = categoriesWithProducts.filter(
-    (category) => category.categoryProducts.length > 0
+export default function YourStore() {
+  const reduxCategories = useAppSelector(
```
Comment (claudiocassimiro) [inline src/components/YourStore/index.tsx:16]:
```suggestion
  const categories = useAppSelector(
```
I think just categories is a good name

File: src/components/YourStore/index.tsx [hunk 4]
```diff
@@ -13,58 +12,14 @@ export interface YourStoreProps {
...
-  };
-
-  const removeCategoryFromList = (categoryToDeleteId: string) => {
-    const newCategoryList = categoriesWithProducts.filter(
-      (category) => category.id !== categoryToDeleteId
-    );
-
-    setCategoriesWithProducts(newCategoryList);
-  };
+  const onlyCategoriesWithProducts = reduxCategories.filter(
```
Comment (claudiocassimiro) [inline src/components/YourStore/index.tsx:20]:
```suggestion
  const onlyCategoriesWithProducts = categories.filter(
```

**Review by claudiocassimiro — APPROVED** [review]

**Review by ebertonArouxa — APPROVED** [review]

**Pull request merged**